cmd/cap1188/main.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	"periph.io/x/periph/conn/gpio"
+	"periph.io/x/periph/conn/gpio/gpioreg"
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/i2c/i2creg"
+	"periph.io/x/periph/conn/pin"
+	"periph.io/x/periph/conn/pin/pinreg"
+	"periph.io/x/periph/experimental/devices/cap1188"
+	"periph.io/x/periph/host"
+)
+
+func mainImpl() error {
+	i2cID := flag.String("i2c", "", "I²C bus to use")
+	i2cADDR := flag.Uint("ia", 0x29, "I²C bus address to use, Pimoroni's Drum Hat is 0x2c")
+	hz := flag.Int("hz", 0, "I²C bus/SPI port speed")
+	verbose := flag.Bool("v", false, "verbose mode")
+	flag.Parse()
+	if !*verbose {
+		log.SetOutput(ioutil.Discard)
+	} else {
+		cap1188.Debug = true
+	}
+	log.SetFlags(log.Lmicroseconds)
+
+	opts := cap1188.DefaultOpts()
+	if *i2cADDR != 0 {
+		opts.Address = uint16(*i2cADDR)
+	}
+
+	if _, err := host.Init(); err != nil {
+		return fmt.Errorf("couldn't init the host - %s", err)
+	}
+
+	var dev *cap1188.Dev
+	i2cBus, err := i2creg.Open(*i2cID)
+	if err != nil {
+		return fmt.Errorf("couldn't open the i2c bus - %s", err)
+	}
+	defer i2cBus.Close()
+	if p, ok := i2cBus.(i2c.Pins); ok {
+		printPin("SCL", p.SCL())
+		printPin("SDA", p.SDA())
+	} else {
+		fmt.Println("i2cBus.(i2c.Pins) failed")
+	}
+
+	if *hz != 0 {
+		if err := i2cBus.SetSpeed(int64(*hz)); err != nil {
+			return fmt.Errorf("couldn't set the i2c bus speed - %s", err)
+		}
+	}
+	// The alert pin is the pin connected to the IRQ/interrupt pin and indicates
+	// when a touch event occurs.
+	alertPin := gpioreg.ByName("GPIO25")
+	if alertPin == nil {
+		return errors.New("invalid alert GPIO pin number")
+	}
+	if err := alertPin.In(gpio.PullUp, gpio.BothEdges); err != nil {
+		return err
+	}
+	if *verbose {
+		fmt.Printf("cap1188: alert pin: %#v\n", alertPin)
+	}
+
+	resetPin := gpioreg.ByName("GPIO21")
+	if resetPin == nil {
+		return errors.New("invalid reset GPIO pin number")
+	}
+	opts.AlertPin = alertPin
+	opts.ResetPin = resetPin
+
+	if dev, err = cap1188.NewI2C(i2cBus, opts); err != nil {
+		return fmt.Errorf("couldn't open cap1188 - %s", err)
+	}
+	time.Sleep(200 * time.Millisecond)
```

Comment:
?

Reply:
<img width="646" alt="cap1188_-_8_channel_capacitive_touch_sensor_with_8_led_drivers_datasheet_-_smsc" src="https://user-images.githubusercontent.com/113/34082904-dda4782c-e31b-11e7-9f44-24b504c36153.png">
The device needs some time to be ready

Reply:
should I move this to the constructor instead?

Reply:
The best would be to note in the constructor when it got initialized and then block in all calls if not enough time passed.

But sleeping in the constructor is fine otherwise.

Reply:
done